---
Author: "Sushmita Maltare"
title: "R Notebook: TCGA_RNAseq_analysis"
output: html_notebook
---

This notebook contains my code and workflow for RNA-seq analysis on TCGA-LGG project which is brain lower grade glioma project containing 516 cases for transcriptome profiling.

First, I am loading following packages required for analysis:
```{r}
# Load packages
library("TCGAbiolinks")
library("limma")
library("edgeR")
library("glmnet")
library("factoextra")
library("FactoMineR")
library("caret")
library("SummarizedExperiment")
library("gplots")
library("survival")
library("survminer")
library("RColorBrewer")
library("gProfileR")
library("genefilter")
```

Let's look at the projects available on GDC:
```{r}
GDCprojects = getGDCprojects()

head(GDCprojects[c("project_id", "name")])
```

I have picked brain lower grade glioma project which has a project id "TCGA-LGG". GDCquery() function from TCGAbiolinks package allows me to look at some of the information available for the TCGA-LGG project available in the TCGA database. I am going to look at the RNAseq data for this project 
```{r}
query_lgg <- GDCquery(project = "TCGA-LGG", data.category = "Transcriptome Profiling", experimental.strategy = "RNA-Seq", workflow.type = "HTSeq - Counts")  
```
getResults() function would give me the counts data in the form of a table which will be useful for further analysis
```{r}
lgg_results <- getResults(query_lgg)
```
```{r}
colnames(lgg_results)
```
```{r}
head(lgg_results$sample_type) #looking at the type of tumor
#subset samples with sample_type entered as "Primary Tumor"
primary_tumor_res <- lgg_results[lgg_results$sample_type == "Primary Tumor",]     
#subset samples which are not entered as "Primary Tumor"
nonprimary_tumor_res <- lgg_results[lgg_results$sample_type != "Primary Tumor",]
nonprimary_tumor_res$sample_type    #Looking at the sample_type in non-primary tumor 

```
We can see above that there are 18 samples with sample_type given as "Recurrent Tumor" and remaining 511 samples have sample_type given as "Primary Tumor". I am removing samples with "Recurrent tumor" as sample_type. Redoing the query:

```{r}
query_lgg <- GDCquery(project = "TCGA-LGG", data.category = "Transcriptome Profiling", experimental.strategy = "RNA-Seq", workflow.type = "HTSeq - Counts", sample.type = "Primary Tumor") 
```
Downloading files specified in the query above:
```{r}
GDCdownload(query = query_lgg)
```
Loading the RNASeq data into R with GDCprepare() function:
```{r}
lgg_data <- GDCprepare(query_lgg)
```
```{r}
dim(lgg_data)     #dim() function gives the dimensions of lgg_data counts table
data.class(lgg_data)  #the lgg_data is loaded as a Ranged summarized experiment as seen below
```
The data is loaded in R as Ranged summarized experiment object. Therefore, we will use three functions, colData(), rowData(), and assay(), to access data from this object. colData() function can access the clinical data table associated with the samples.
```{r}
colnames(colData(lgg_data))
```

assay() function is used to access counts data for the summarized experiment object. Rows are gene ids and columns are sample names in the counts table. 
```{r}
head(assay(lgg_data)[,1:5])       #counts for first 6 genes and first 5 samples
```

Saving RNASeq data as '.RDS' file which can be loaded anytime using readRDS() function without having to run the whole notebook again
```{r}
saveRDS(object = lgg_data,
        file = "lgg_data.RDS",
        compress = FALSE)
```
Let's load the lgg_data.RDS:
```{r}
lgg_data <- readRDS(file = "lgg_data.RDS")
```

I am going to look the summaries of some columns from colData to get a better understanding. I will use table() function for that:
```{r}
table(lgg_data@colData$gender)
```
There are a total of 510 patients with a gender specified, and 1 patient whose gender is not specified, we will remove that sample from counts table:
```{r}
complete.cases(lgg_data@colData$gender)

phenotype_data[366, c("patient", "gender")] #366th row has NA as gender
```
```{r}
NA_removed_pheno <- colData(lgg_data)[-366,]
NA_removed_counts <- subset(assay(lgg_data), select = -366)       #removing 366th column which corresponds to the sample with NA as
```

```{r}
dim(NA_removed_pheno)
dim(NA_removed_counts)
```

```{r}
table(lgg_data@colData$paper_Age..years.at.diagnosis.)
```
```{r}
table(lgg_data@colData$paper_Vital.status..1.dead.)
```
```{r}
table(lgg_data@colData$tumor_stage)
```
```{r}
table(lgg_data@colData$race)
```
```{r}
table(lgg_data@colData$definition)
```

Differential Expression analysis:

I will be looking at whether there are differentially expressed genes in male vs female brain glioma patients. First, I will normalize the counts data in order to handle batch effects: 

```{r}
phenotype_data <- colData(lgg_data)
gender_factor <- factor(phenotype_data$gender) #converting gender into a factor variable in order to use it for limma package
gender_factor <- relevel(gender_factor, ref = "female") #setting "female" as the reference level
```

```{r}
design_mat <- model.matrix(~gender_factor)  #creating a design matrix to be used for DE analysis
head(design_mat)
```

```{r}
dge_gender <- DGEList(counts = NA_removed_counts, samples = NA_removed_pheno, genes = as.data.frame(rowData(lgg_data)))  #creating a DGEList object from lgg_data, facilitates filtering

keep_genes <- filterByExpr.default(dge_gender, design_mat)   #keeps genes which have read counts above 10 (default)
dge_gender <- dge_gender[keep_genes, , keep.lib.sizes = FALSE]   #filtering the DGEList object
rm(keep_genes)
dim(dge_gender)
```
I am using Trimmed mean of M-values (TMM) method for data normalization. And I am using VOOM method to convert the data to have a similar variance as arrays (must do this for limma)
```{r}
dge_gender = calcNormFactors(dge_gender,method="TMM")
v_gender = voom(dge_gender,design_mat,plot=TRUE)
```
I am going to run lmFit() function on the v_gender object obtained above and then eBayes() function on the linear models obtained from lmFit(). I will use eBayes results statistics to rank the differentially expressed genes:
```{r}
linear_mods <- lmFit(v_gender, design_mat)
linear_mods <- eBayes(linear_mods)
```
 Let's look at the top 10 differentially expressed genes:
```{r}
DE_genes <- topTable(linear_mods, coef = 1, sort.by ="p")
print(DE_genes)
```






